doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -687,13 +687,126 @@ side. E.g.:
    Notation "'apply_id' f a1 .. an" := (.. (f a1) .. an)
      (at level 10, f ident, a1, an at level 9).
 
+Custom entries
+~~~~~~~~~~~~~~
+
+.. cmd:: Declare Custom Entry @ident
+
+New term and pattern grammar entries, called *custom entries*, can be
+declared using the :n:`Declare Custom Entry @ident` command and later
+referred to using the entry name :n:`custom @ident`. For instance, we
+may want to define an ad hoc parser for arithmetical operations and
+proceed as follows:
+
+.. coqtop:: all
+
+   Inductive Expr :=
+   | One : Expr
+   | Mul : Expr -> Expr -> Expr
+   | Add : Expr -> Expr -> Expr.
+
+   Declare Custom Entry expr.
+   Notation "[ e ]" := e (e custom expr at level 2).
+   Notation "1" := One (in custom expr at level 0).
+   Notation "x y" := (Mul x y) (in custom expr at level 1, left associativity).
+   Notation "x + y" := (Add x y) (in custom expr at level 2, left associativity).
+   Notation "( x )" := x (in custom expr, x at level 2).
+   Notation "{ x }" := x (in custom expr, x constr).
+   Notation "x" := x (in custom expr at level 0, x ident).
+
+   Axiom f : nat -> Expr.
+   Check fun x y z => [1 + y z + {f x}].
+   Unset Printing Notations.
+   Check fun x y z => [1 + y z + {f x}].
+   Set Printing Notations.
+   Check fun e => match e with
+   | [1 + 1] => [1]
+   | [x y + z] => [x + y z]
+   | y => [y + e]
+   end.
+
+Custom entries have levels, like the main grammar of terms and grammar
+of patterns have. The lower level is 0 and this is the level used by
+default to put rules delimited with tokens on both ends. The level is
+left to be inferred by Coq when using :n:`in custom @ident``. The
+level is otherwise given explicitly by using the syntax :n:`in custom
+@ident at level @num`, where :n:`@num` refers to the level.
```

Comment:
You can use ```:token:`num` ``` instead of ```:n:`@num` ```. See https://github.com/coq/coq/tree/master/doc/sphinx#coq-roles

Reply:
I tried both version but couldn't understand the difference. They both seems to link to where the entry is defined in the manual, and the only difference seems to be a color blue vs a color green.

Reply:
Yes, I'm not exactly sure either.